src/fielddef.ts

```diff
@@ -147,7 +147,8 @@ export function field(fieldDef: FieldDef, opt: FieldRefOption = {}) {
 
     if (!opt.nofn) {
       if (fieldDef.bin) {
-        fn = 'bin';
+        const bin: Bin = isBoolean(fieldDef.bin) ? {} : {...fieldDef.bin};
```

Comment:
Actually, you need a separate method for that to make it correct inside transform.  Let's make a `getBinDef(bin: boolean | Bin): Bin`.